matchms/filtering/derive_adduct_from_name.py

```diff
@@ -0,0 +1,48 @@
+from ..typing import SpectrumType
+from ..utils import looks_like_adduct
+
+
+def derive_adduct_from_name(spectrum_in: SpectrumType, remove_adduct_from_name=True) -> SpectrumType:
+    """Find adduct in compound name and add to metadata (if not present yet).
+
+    Method to interpret the given compound name to find the adduct.
+    Args:
+    ----
+    spectrum_in: SpectrumType
+        Input spectrum.
+    remove_adduct_from_name: bool
+        Remove found adducts from compound name if set to True. Default is True.
+    """
+    if spectrum_in is None:
+        return None
+
+    spectrum = spectrum_in.clone()
+
+    # Get compound name
```

Comment:
You have duplicated this piece of code quite a lot. Could you extract it as a function and place it in some utils file or find some other way to get rid of the duplication?

Reply:
It still felt a bit too "small" to me to add it to utils. I removed the unnecessary comments and kept it unchanged otherwise. Would it be OK for you to keep it at that for now?